src/main/java/com/github/dockerjava/core/command/CreateContainerCmdImpl.java

```diff
@@ -125,11 +139,50 @@ public CreateContainerCmdImpl(CreateContainerCmd.Exec exec, String image) {
      */
     @Override
     public CreateContainerResponse exec() throws NotFoundException, ConflictException {
+        //code flow taken from https://github.com/docker/docker/blob/master/runconfig/opts/parse.go
```

Comment:
So what is your suggestion? Completely remove the CreateContainerCmd#withLinks method and expose the HostConfig and ContainerNetwork directly? And how you would like to expose the ContainerNetwork? CreateContainerCmd#withContainerNetwork() will not work if i follow your guideline, since it must be set on an Map under a key which has the same name as the user defined network. In addition using any other keys on this Map has no effect. 

The API is really not developer friendly and it is not understandable right form the API when to use HostConfig#withLinks or ContainerNetwork#withLinks. And since network API is the great new feature of docker it should get a good usable API.